dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -285,20 +282,50 @@ private void checkSinglePermission(PermissionListener listener, String permissio
     checkMultiplePermissions(adapter, Collections.singleton(permission), thread);
   }
 
-  private void checkMultiplePermissions(MultiplePermissionsListener listener,
-      Collection<String> permissions, Thread thread) {
+  private void checkMultiplePermissions(final MultiplePermissionsListener listener,
+      final Collection<String> permissions, Thread thread) {
     checkNoDexterRequestOngoing();
     checkRequestSomePermission(permissions);
 
     pendingPermissions.clear();
     pendingPermissions.addAll(permissions);
     multiplePermissionsReport.clear();
     this.listener = new MultiplePermissionListenerThreadDecorator(listener, thread);
-
-    startTransparentActivityIfNeeded();
+    if (!everyPermissionIsGranted(permissions) || !contextIsInstanceOfActivity()) {
```

Comment:
Just a matter of taste but I feel using `!A || !B` is harder to read than `A && B` and reverse the order of the `if/else` bodies, feel free to keep it this way if you want though